airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry.py

```diff
@@ -104,14 +137,35 @@ def generate_and_persist_registry(
     )
 
     registry_dict = {"sources": [], "destinations": []}
-    for blob in registry_entry_file_blobs:
-        connector_type, registry_entry = read_registry_entry_blob(blob)
-        plural_connector_type = f"{connector_type}s"
 
-        # We santiize the registry entry to ensure its in a format
+    docker_repository_to_rc_metadata_entry = {
+        release_candidate_metadata_entries.metadata_definition.data.dockerRepository: release_candidate_metadata_entries
+        for release_candidate_metadata_entries in release_candidate_metadata_entries
+    }
+    docker_repository_to_rc_registry_entry = {
+        release_candidate_registry_entries.dockerRepository: release_candidate_registry_entries
+        for release_candidate_registry_entries in release_candidate_registry_entries
+    }
+
+    for latest_registry_entry in latest_registry_entries:
+        connector_type = get_connector_type_from_registry_entry(latest_registry_entry)
+        plural_connector_type = f"{connector_type.value}s"
+
+        # We sanitize the registry entry to ensure its in a format
         # that can be parsed by pydantic.
-        registry_entry_dict = to_json_sanitized_dict(registry_entry)
-        enriched_registry_entry_dict = apply_metrics_to_registry_entry(registry_entry_dict, connector_type, latest_connnector_metrics)
+        registry_entry_dict = to_json_sanitized_dict(latest_registry_entry)
+        enriched_registry_entry_dict = apply_metrics_to_registry_entry(registry_entry_dict, connector_type, latest_connector_metrics)
+        if (
+            latest_registry_entry.dockerRepository in docker_repository_to_rc_metadata_entry
+            and latest_registry_entry.dockerRepository in docker_repository_to_rc_registry_entry
```

Comment:
💅 📚 I think we'd benefit from a `apply_release_candidate_entries` function to hold this logic